File: app/src/main/java/com/emojimixer/functions/RequestNetwork.java
package com.emojimixer.functions;

import android.app.Activity;

import java.util.HashMap;

public class RequestNetwork {
    private final HashMap<String, Object> params = new HashMap<>();
    private HashMap<String, Object> headers = new HashMap<>();
    private int requestType = 0;

    public RequestNetwork() {
    }

    public HashMap<String, Object> getParams() {
        return params;
    }

    public HashMap<String, Object> getHeaders() {
        return headers;
    }

    public void setHeaders(HashMap<String, Object> headers) {
        this.headers = headers;
    }

    public int getRequestType() {
        return requestType;
    }

    public void startRequestNetwork(
            String method,
            String url,
            String tag,
            RequestListener requestListener
    ) {
        RequestNetworkController.getInstance().execute(this, method, url, tag, requestListener);
    }

    public interface RequestListener {
        void onResponse(String tag, String response, HashMap<String, Object> responseHeaders);
        void onErrorResponse(String tag, String message);
    }
}

File: app/src/main/java/com/emojimixer/functions/RequestNetwork.kt
package com.emojimixer.functions

import android.app.Activity

class RequestNetwork(val activity: Activity) {
    var params = HashMap<String, Any>()
        private set
    var headers = HashMap<String, Any>()
    var requestType = 0
        private set

    fun startRequestNetwork(
        method: String?,
        url: String?,
        tag: String?,
        requestListener: RequestListener?
    ) {
        RequestNetworkController.getInstance().execute(this, method, url, tag, requestListener)
    }

    interface RequestListener {
        fun onResponse(tag: String?, response: String?, responseHeaders: HashMap<String?, Any?>?)
        fun onErrorResponse(tag: String?, message: String?)
    }
}